pkg/controllers/externalsecret/externalsecret_controller.go

```diff
@@ -198,8 +209,13 @@ func (r *Reconciler) getProviderSecretData(ctx context.Context, providerClient p
 }
 
 func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
+	// Skip running Reconcile() when there are NO changes to the .spec field
+	// Useful to not trigger Reconcile() when only updating the object's status/metadata
+	p := predicate.GenerationChangedPredicate{}
```

Comment:
Reconciling on status seems odd at first glance, but because this is the managing controller for the resource we want to reconcile on any change to the resource. Otherwise if another system modifies or clears the status this controller will not react and set the status correctly. I would recommend that we reconcile on all changes and not just generation changes.